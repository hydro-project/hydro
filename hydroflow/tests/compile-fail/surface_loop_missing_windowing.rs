fn main() {
    let mut df = hydroflow::dfir_syntax! {
        a = source_iter(0..10);
        loop {
            a -> null();
        }
    };
    df.run_available();
}
