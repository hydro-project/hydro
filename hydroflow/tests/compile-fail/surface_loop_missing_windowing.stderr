warning: `loop` blocks are not yet supported.
 --> tests/compile-fail/surface_loop_missing_windowing.rs:4:9
  |
4 |         loop {
  |         ^^^^

error: Operator `null(...)` entering a loop context must be a windowing operator, but is not.
 --> tests/compile-fail/surface_loop_missing_windowing.rs:5:18
  |
5 |             a -> null();
  |                  ^^^^^^
